install.sh

```diff
@@ -0,0 +1,448 @@
+    if [[ "$RELEASE_MAJOR_VERSION_NUMBER" -gt "$CURRENT_MAJOR_VERSION_NUMBER" ]]; then
+      return 0
+    elif [[ "$RELEASE_MAJOR_VERSION_NUMBER" -eq "$CURRENT_MAJOR_VERSION_NUMBER" ]]; then
+      if [[ "$RELEASE_MINOR_VERSION_NUMBER" -gt "$CURRENT_MINOR_VERSION_NUMBER" ]]; then
+        return 0
+      elif [[ "$RELEASE_MINOR_VERSION_NUMBER" -eq "$CURRENT_MINOR_VERSION_NUMBER" ]]; then
+        if [[ "$RELEASE_MINIMUM_VERSION_NUMBER" -gt "$CURRENT_MINIMUM_VERSION_NUMBER" ]]; then
+          return 0
+        else
+          return 1
+        fi
+      else
+        return 1
+      fi
+    else
+      return 1
+    fi
+  elif [[ "$RELEASE_VERSION" == "$CURRENT_VERSION" ]]; then
+    return 1
+  fi
+}
+
+download_hysteria() {
+  DOWNLOAD_LINK="https://github.com/HyNetwork/hysteria/releases/download/$RELEASE_VERSION/hysteria-linux-$MACHINE"
+  echo "Downloading Hysteria archive: $DOWNLOAD_LINK"
+  if ! curl -x "${PROXY}" -R -H 'Cache-Control: no-cache' -o "$BIN_FILE" "$DOWNLOAD_LINK"; then
+    echo 'error: Download failed! Please check your network or try again.'
+    return 1
+  fi
+}
+
+install_file() {
+  NAME="$1"
+  if [[ "$NAME" == "hysteria-linux-$MACHINE" ]] ; then
+    install -m 755 "${TMP_DIRECTORY}/$NAME" "/usr/local/bin/hysteria"
+  fi
+}
+
+install_hysteria() {
+  # Install hysteria binary to /usr/local/bin/
+  install_file hysteria-linux-$MACHINE
+  
+  # Install hysteria configuration file to $JSON_PATH
+  # shellcheck disable=SC2153
+  if [[ -z "$JSONS_PATH" ]] && [[ ! -d "$JSON_PATH" ]]; then
+    install -d "$JSON_PATH"
+    echo -e '{\n"listen": ":36712",\n "acme": {\n"domains": [\n\n"your.domain.com"\n],\n"email": "hacker@gmail.com"\n},\n"obfs": "fuck me till the daylight",\n"up_mbps": 100,\n"down_mbps": 100\n}' | python -m json.tool > "${JSON_PATH}/config.json"
+    CONFIG_NEW='1'
+  fi
+
+  # Install hysteria configuration file to $JSONS_PATH
+  if [[ -n "$JSONS_PATH" ]] && [[ ! -d "$JSONS_PATH" ]]; then
+    install -d "$JSONS_PATH"
+    echo -e '{\n"listen": ":36712",\n "acme": {\n"domains": [\n\n"your.domain.com"\n],\n"email": "hacker@gmail.com"\n},\n"obfs": "fuck me till the daylight",\n"up_mbps": 100,\n"down_mbps": 100\n}' | python -m json.tool > "${JSONS_PATH}/${BASE}.json" 
+    CONFDIR='1'
+  fi
+}
+
+install_startup_service_file() {
+  echo "[Unit]
+Description=Hysteria, a feature-packed network utility optimized for networks of poor quality
+Documentation=https://github.com/HyNetwork/hysteria/wiki
+After=network-online.target nss-lookup.target
+Wants=network-online.target systemd-networkd-wait-online.service
+
+[Service]
+User=root
+CapabilityBoundingSet=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
+AmbientCapabilities=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
+NoNewPrivileges=true
+ExecStart=/usr/local/bin/hysteria -c /etc/hysteria/config.json server
+Restart=on-failure
+RestartPreventExitStatus=1
+RestartSec=5
+
+[Install]
+WantedBy=multi-user.target" > /lib/systemd/system/hysteria.service
```

Comment:
Please install the hysteria server systemd unit as hysteria-server.service and hysteria-server@.service, v2ray don't have a split command for server but hysteria do.